src/wp-includes/l10n/class-wp-translation-controller.php

```diff
@@ -0,0 +1,410 @@
+	 * @param string                      $textdomain Optional. Text domain. Default 'default'.
+	 * @param string                      $locale     Optional. Locale. Default current locale.
+	 * @return string|false Translation on success, false otherwise.
+	 */
+	public function translate_plural( array $plurals, int $number, string $context = '', string $textdomain = 'default', string $locale = null ) {
+		if ( '' !== $context ) {
+			$context .= "\4";
+		}
+
+		$text        = implode( "\0", $plurals );
+		$translation = $this->locate_translation( "{$context}{$text}", $textdomain, $locale );
+
+		if ( false === $translation ) {
+			$text        = $plurals[0];
+			$translation = $this->locate_translation( "{$context}{$text}", $textdomain, $locale );
+
+			if ( false === $translation ) {
+				return false;
+			}
+		}
+
+		/* @var WP_Translation_File $source */
+		$source = $translation['source'];
+		$num    = $source->get_plural_form( $number );
+
+		// See \Translations::translate_plural().
+		return $translation['entries'][ $num ] ?? $translation['entries'][0];
+	}
+
+	/**
+	 * Returns all existing headers for a given text domain.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $textdomain Optional. Text domain. Default 'default'.
+	 * @return array<string, string> Headers.
+	 */
+	public function get_headers( string $textdomain = 'default' ) {
+		if ( array() === $this->loaded_translations ) {
+			return array();
+		}
+
+		$headers = array();
+
+		foreach ( $this->get_files( $textdomain ) as $moe ) {
+			foreach ( $moe->headers() as $header => $value ) {
+				$headers[ $this->normalize_header( $header ) ] = $value;
+			}
+		}
+
+		return $headers;
+	}
+
+	/**
+	 * Normalizes header names to be capitalized.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $header Header name.
+	 * @return string Normalized header name.
+	 */
+	protected function normalize_header( string $header ) {
+		$parts = explode( '-', $header );
+		$parts = array_map( 'ucfirst', $parts );
+		return implode( '-', $parts );
+	}
+
+	/**
+	 * Returns all entries for a given text domain.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $textdomain Optional. Text domain. Default 'default'.
+	 * @return array<string, string> Entries.
+	 */
+	public function get_entries( string $textdomain = 'default' ) {
+		if ( array() === $this->loaded_translations ) {
+			return array();
+		}
+
+		$entries = array();
+
+		foreach ( $this->get_files( $textdomain ) as $moe ) {
+			$entries = array_merge( $entries, $moe->entries() );
+		}
+
+		return $entries;
+	}
+
+	/**
+	 * Locates translation for a given string and text domain.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $singular   Singular translation.
+	 * @param string $textdomain Optional. Text domain. Default 'default'.
+	 * @param string $locale     Optional. Locale. Default current locale.
+	 * @return array{source: WP_Translation_File, entries: string[]}|false Translations on success, false otherwise.
```

Comment:
Another array shape (❤️). Perhaps add the `@type` convention as well? cf. https://wordpress.slack.com/archives/C5VCTJGH3/p1705428874962969